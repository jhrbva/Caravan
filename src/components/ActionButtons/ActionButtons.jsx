import React from 'react';
import axios from 'axios';

import './ActionButtons.scss';
import { Link } from 'react-router-dom';
import ModalButton from '../ModalButton/ModalButton';
import { withRouter } from 'react-router';

class ActionButtons extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			accepted: this.props.accepted,
		};
	}

	responseToInvitation = () => {
		axios
			.post('/invitations/accept', {
				userid: this.props.userid,
				tripid: this.props.tripid,
				accepted: this.state.accepted,
			})
			.then(function (response) {
				console.log(response);
			})
			.catch(function (error) {
				console.log(error);
			});
	};

	requestChangeButton = () => {
		const { history } = this.props;
		return history.push({
			pathname: '/requestchange',
			tripid: this.props.tripid,
		});
	};

	renderInitial = () => {
		return (
			<div>
				<ModalButton
					value={'Accept'}
					onClick={() => {
						this.setState({ accepted: true }, function () {
							this.responseToInvitation();
						});
					}}
					numberofbuttons={'three'}
				/>
				<ModalButton
					value={'Reject'}
					onClick={() => {
						this.setState({ accepted: false }, function () {
							this.responseToInvitation();
						});
					}}
					numberofbuttons={'three'}
				/>
				<ModalButton
					value={'Request Change'}
					onClick={() => {
						this.requestChangeButton();
					}}
					numberofbuttons={'three'}
				/>
			</div>
		);
	};

	renderAccepted = () => {
		return (
			<div>
				<Link to='/map'>
					<ModalButton value={'Start Trip'} additionalClass={'start'} />
				</Link>
				<ModalButton
					value={'Leave Trip'}
					onClick={() => {
						this.setState({ accepted: false }, function () {
							this.responseToInvitation();
						});
					}}
					additionalClass={'leave'}
				/>
			</div>
		);
	};

	renderRejected = () => {
		return (
			<div>
				<ModalButton
					value={'Accept Invitation'}
					onClick={() => {
						this.setState({ accepted: true }, function () {
							this.responseToInvitation();
						});
					}}
				/>
				<ModalButton
					value={'Request a Change'}
					onClick={() => {
						this.requestChangeButton();
					}}
				/>
			</div>
		);
	};

	render() {
		if (this.state.accepted === true) {
			return this.renderAccepted();
		} else if (this.state.accepted === false) {
			return this.renderRejected();
		} else return this.renderInitial();
	}
}
export default withRouter(ActionButtons);
