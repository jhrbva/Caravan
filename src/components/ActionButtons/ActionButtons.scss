@import '../../utilities/colorPalette.scss';

.invitation-btns.one-btn {
	.big-button {
		width: 100%;
	}
}
.invitation-btns.two-btns {
	.big-button {
		width: 50%;
	}
}
.invitation-btns.three-btns {
	.big-button {
		width: 33.333333%;
	}
}

.rejected-msg {
	p {
		margin: 10px 0;
		text-align: center;
	}
}

.invitation-btns .big-button:active,
.invitation-btns .big-button:focus {
	background: $fun-yellow;
}
