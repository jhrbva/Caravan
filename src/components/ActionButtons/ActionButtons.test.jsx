import React from 'react';
import { shallow } from 'enzyme';
import ActionButtons from './ActionButtons';

describe('Action Buttons', () => {
	it('should match the snapshot', () => {
		const wrapper = shallow(<ActionButtons />);
		expect(wrapper).toMatchSnapshot();
	});
});
