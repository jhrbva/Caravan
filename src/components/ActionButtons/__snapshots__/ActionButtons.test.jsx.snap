// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Buttons should match the snapshot 1`] = `
<div
  className="invitation-btns"
>
  <BigButton
    onClick={[Function]}
    value="Accept"
  />
  <BigButton
    onClick={[Function]}
    value="Reject"
  />
  <BigButton
    value="Request Change"
  />
</div>
`;
